_pages/algorithm-development/languages/python.md

```diff
@@ -10,57 +10,90 @@ image:
 
-If you've followed the <a href="{{site.baseurl}}/algorithm-development/algorithm-basics/your-first-algo">Getting Started Guide</a>, you'll notice in your algorithm editor, there is boilerplate code that returns "Hello" and whatever you input to the console.
+Let's look at an example to clarify some of these concepts.
 
-The main thing to note about the algorithm is that it's wrapped in the apply() function.
+## Hello World
 
-The apply() function defines the input point of the algorithm. We use the apply() function in order to make different algorithms standardized. This makes them easily chained and helps authors think about designing their algorithms in a way that makes them easy to leverage and predictable for end users.
-{: .notice-info}
+Below you'll find a `src/Algorithm.py` file which prints "hello" plus an input when it is invoked. We start by importing the Algorithmia ADK, and then defining our `apply` function, followed by our call to the handler function `ADK()`, and finally calling `init()` to start the function.
 
-Go ahead and remove the boilerplate code below that's inside the apply() function because we'll be writing a different algorithm in this tutorial:
+{% highlight python %}
+from Algorithmia import ADK
 
-<img src="{{site.cdnurl}}{{site.baseurl}}/images/post_images/algo_dev_lang/algorithm_console_python.png" alt="Algorithm console Python" class="screenshot">
+def apply(input):
+    return "hello {}".format(str(input))
 
+algorithm = ADK(apply)
+algorithm.init("Algorithmia")
+{% endhighlight %}
 
-## Managing Dependencies
+When executed on the Algorithmia platform and providing the string "HAL 9000" as an input, this algorithm will output "hello HAL 9000". If executed locally, the algorithm will print `hello Algorithmia` to `stdout` instead.
 
-Algorithmia supports adding 3rd party dependencies via the <a href="https://pypi.python.org/pypi">Python Package Index (PyPI)</a> using a requirements.txt file.
+## Loaded State
 
-On the algorithm editor page there is a button on the top right that says "Dependencies". Click that button and you'll see a modal window:
+Developing with the ADK allows you to make use of an optional `load()` function for preparing an algorithm for runtime operations. Using `load()`, we can load state into memory prior to a function's execution and then pass that data to `apply()` by adding an additional `globals` parameter in the `apply()` function. 
```

Comment:
Again, adding a "why" to these explanations would be helpful - why/use case to use the load() func and load state into memory before execution of the func

Reply:
Happy to add more info, but can't answer "why" for this one. @zeryx Any thoughts on this?

Reply:
clearer understanding of when and how to to get and initialize your algorithms model dependencies in one; clearly defined place. Plus, it will also enable the ability for track loading characteristic metrics separately from runtime characteristics in the near future, without needing to change or upgrade your algorithm code.

Reply:
Something like:
"When you load state outside of the `apply()` function, it remains in memory and is available for subsequent calls to the algorithm. This enhances the performance of the algorithm, because data does not need to be read from disk for every algorithm execution; instead, it is accessed from memory, which is orders of magnitude faster. To learn more, see our documentation on [Leveraging Global Scope for Performance](https://training.algorithmia.com/enterprise-20-2-using-data-sources/688899#loading-persistent).

Reply:
We don't recommend using global scope anymore, that was a hack that we used for a while to enable 1 time loading and initialization of algorithm state; we now recommend using a `load()` function which is a great location to define any initialization process. If something fails during loading, in the past you would receive no error message; this made it really hard to debug load time failures. Now, if a failure occurs during loading, those exceptions get raised as `loadingError`s!

Reply:
OK, I've updated this with some of the copy from the page @lemonez referenced, but I left out a link to that page until it can be updated to reflect the use of `load()` rather than making use of global scope.

Reply:
Good call, sounds like a great approach. I've made a note to add a link here once that content is updated.